### PR TITLE
ci: Fix CI builders

### LINKED ISSUE #414: m1 mac on github actions
Seems this has been released today and ci-build-wheel supports it too https://github.com/pypa/cibuildwheel/releases/tag/v2.16.5

### PR DESCRIPTION
**Pull request by LecrisUT**
- [x] Enable all macos/windows buidlers
- [x] Fix windows builders
- [ ] Fix Macos builders. Maybe we just skip `macos-12`: https://github.com/spglib/spglib/pull/459#issuecomment-2023171931
- [x] Update `ci-build-wheel` to include M1 variants: Running all configurations at https://github.com/LecrisUT/spglib/actions/runs/8456467424

Depends-on: #431 

Closes #414

### CODE REVIEW

**Comment by LecrisUT** [comment]
Interesting, only `macos-12` a.k.a. `macos-latest` has the issue of `test_with_broken_symmetry`. For now, are we ok with just dropping that one since we would probably want to focus on `macos-14` which has M1 chip. @atztogo @lan496

~~The other issues with the windows machines are fixable though since it seems they are only testing issues remaining, I'll try to handle those.~~ Done.

Question is, do we want to include this one in 2.4 as well? This one has a minor fix to #233 on a7c63da5350df2c29ff97ada09cf83e412c9ad83, but otherwise it's completely neutral (can be added at any point). It would make PyPI have more MacOS wheels, so might be nice :wink:

File: .github/workflows/step_build-wheel.yaml [hunk 1]
```diff
@@ -43,11 +33,9 @@ jobs:
         with:
           platforms: all
       - name: Build wheels
-        uses: pypa/cibuildwheel@v2.16
+        uses: pypa/cibuildwheel@v2.17
         env:
           CIBW_BUILD: ${{ inputs.cibw_build }}
-          CIBW_ARCHS_LINUX: ${{ inputs.cibw_archs_linux }}
-          CIBW_ARCHS_MACOS: ${{ inputs.cibw_archs_macos }}
```
Comment (lan496) [inline .github/workflows/step_build-wheel.yaml:50]: Does it mean we do not need to manually specify `CIBW_ARCHS_LINUX` and `CIBW_ARCHS_MACOS` anymore?

Comment (LecrisUT) [reply .github/workflows/step_build-wheel.yaml:50]: Yes, we are only using native builders anyway and with `macOS-14` there is M1-arm native builders

**Review by atztogo — APPROVED** [review]

File: .github/workflows/step_build-wheel.yaml (hunk 1 above)
Comment (LecrisUT) [reply .github/workflows/step_build-wheel.yaml:50]: Oh, wait the linux arch is not picked up :/

Edit: now it's picked up

**Review by lan496 — APPROVED** [review]

**Pull request merged**